### PR TITLE
[W-5295461] add response validation

### PR DESCRIPTION
**Pull request by cassiodias**
* added test to validate invalid schema
* added testUtil method to validate schema
* added gitignore

### CODE REVIEW

File: connection/testConnection.js [hunk 1]
```diff
@@ -9,13 +9,12 @@ const tu = require('../utils/testUtils');
...
   before(tu.buildConnection);
 
-  const ctx = {
-    baseTrustUrl: 'https://trust.salesforce.com/api',
-    statusLinkUrl: 'http://status.salesforce.com/status',
-  };
-
   describe('prepareUrl >', () => {
     it('prepareUrl, default window', () => {
+      const ctx = {
```
Comment (cassiodias) [inline connection/testConnection.js]: Test was failing on master, one test was affecting the other (context was incorrect bcz aspect was not defined).
Duplicated the ctx and it works, didnt spent time to that.

File: transform/transform.js [hunk 2]
```diff
@@ -142,4 +142,17 @@ module.exports = {
       description: 'An object which maps each Trust1 status enum value to a sample value, messageCode and messageBody.',
     },
   },
+  responseSchema: {
+    type: 'object',
+    required: ['body'],
+    properties: {
+      body: {
+        type: 'array',
+        minItems: 1,
```
Comment (cassiodias) [inline transform/transform.js]: There are no details about the restriction, we can go further on required attributes. 
As I am not sure just added a minimum of one object inside the array.

File: transform/transform.js [hunk 3]
```diff
@@ -142,4 +142,17 @@ module.exports = {
...
+        items: {
+          type: 'object'
```
Comment (iamigo) [inline transform/transform.js]: no schema for the items?

File: transform/testTransform.js [hunk 4]
```diff
@@ -154,6 +154,25 @@ describe('transform tests >', () => {
...
+              "releaseNumber": "210.11.4"
+            }
+          ]
+        };
+        [1, '', {}, { body: { a: '' } }, { body: [1,2] }, { body: [1] }, { body: [] },
+          withoutkey].forEach((it) => {
+            expect(() => tu.validateResponse(it)).to.throw();
+        });
+      });
+    });
```
[the remark this answers is not in the document]
Comment (jgraff2) [reply transform/testTransform.js]: also should test props of the wrong type.

[the remark this answers is not in the document]
Comment (cassiodias) [reply transform/testTransform.js]: "Should also make sure it doesn't throw for a valid response." this has been tested by the rest of the transform executions, no?

**Pull request merged**